public/app/features/plugins/extensions/registry/Registry.ts

```diff
@@ -52,4 +68,13 @@ export abstract class Registry<TRegistryValue, TMapType> {
   getState(): Promise<RegistryType<TRegistryValue>> {
     return firstValueFrom(this.asObservable());
   }
+
+  // Returns a read-only version of the registry.
+  readOnly() {
```

Comment:
I'm not sure this is feasible, but if default is to pass a readonly registry to apps, can we return a readonly interface from here and pass that type down via the context rather than than this class? 
```
export interface ReadOnlyRegistry<TRegistryValue, TMapType> {
  asObservable(): Observable<RegistryType<TRegistryValue>>;
  ...
}
```